ql.go

```diff
@@ -516,6 +528,33 @@ func (r *whereRset) planUnaryOp(x *unaryOperation) (plan, error) {
 }
 
 func (r *whereRset) plan(ctx *execCtx) (plan, error) {
+	if r.sel != nil {
+		var exists bool
+		p, err := r.sel.plan(ctx)
+		if err != nil {
+			return nil, err
+		}
+		err = p.do(ctx, func(i interface{}, data []interface{}) (bool, error) {
+			if len(data) > 0 {
```

Comment:
Thanks for the clarification. Not I think choosing to evaluate only one row is enough for this case.